Project 1/linked_list.cpp

```diff
@@ -4,7 +4,43 @@
 
 LinkedListNode* merge(LinkedListNode* head1, LinkedListNode* head2)
 {
-    return nullptr;
+    LinkedListNode *result;
```

Comment:
Interestingly, we never reach the case when we have to use this garbage `result` value since the only two exit conditions are when either of the heads is `null`. So we replace the `*current` value, which is always equal to this garbage value of the `result`, with either `head1` or `head2`.